### PR TITLE
add api feature request issue template form.

### PR DESCRIPTION
**Pull request by a1012112796**
main aim: want auto attach `kind/api` label to them :)
just a propsal, will close it soon if no one like it. thanks!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/api-request.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- type: dropdown
+  id: github-exist
+  attributes:
+    label: whether github has similar api?
+    description: |
+      If so, please provide a URL in the Description field.
+      github api document: https://docs.github.com/en/rest
+    options:
+    - "Yes"
+    - "No"
```
Comment (JakobDev) [inline .github/ISSUE_TEMPLATE/api-request.yml:21]: Why no checkbox?

File: .github/ISSUE_TEMPLATE/api-request.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: api feature Request
+description: Got an idea for an api that Gitea doesn't have currently?  Submit your idea here!
```
Comment (yardenshoham) [inline .github/ISSUE_TEMPLATE/api-request.yml:2]:
```suggestion
description: Got an idea for an API endpoint that Gitea doesn't have? Submit your idea here!
```

File: .github/ISSUE_TEMPLATE/api-request.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  attributes:
+    value: |
+      1. Please speak English, this is the language all maintainers can speak and write.
+      2. Please ask questions or configuration/deploy problems on our Discord
+         server (https://discord.gg/gitea) or forum (https://discourse.gitea.io).
+      3. Please take a moment to check that your feature hasn't already been suggested.
+- type: dropdown
+  id: github-exist
+  attributes:
+    label: whether github has similar api?
```
Comment (yardenshoham) [inline .github/ISSUE_TEMPLATE/api-request.yml:15]:
```suggestion
    label: whether GitHub has a similar API endpoint?
```

File: .github/ISSUE_TEMPLATE/api-request.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+description: Got an idea for an api that Gitea doesn't have currently?  Submit your idea here!
+labels: ["kind/api", "kind/feature", "kind/proposal"]
+body:
+- type: markdown
+  attributes:
+    value: |
+      1. Please speak English, this is the language all maintainers can speak and write.
+      2. Please ask questions or configuration/deploy problems on our Discord
+         server (https://discord.gg/gitea) or forum (https://discourse.gitea.io).
+      3. Please take a moment to check that your feature hasn't already been suggested.
```
Comment (delvh) [inline .github/ISSUE_TEMPLATE/api-request.yml:11]:
```suggestion
      3. Please take a moment to check that your endpoint hasn't been suggested already.
```

File: .github/ISSUE_TEMPLATE/api-request.yml (hunk 3 above)
Comment (delvh) [reply .github/ISSUE_TEMPLATE/api-request.yml:15]:
```suggestion
    label: Does GitHub have a similar API endpoint?
```

File: .github/ISSUE_TEMPLATE/api-request.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+- type: dropdown
+  id: github-exist
+  attributes:
+    label: whether github has similar api?
+    description: |
+      If so, please provide a URL in the Description field.
+      github api document: https://docs.github.com/en/rest
```
Comment (delvh) [inline .github/ISSUE_TEMPLATE/api-request.yml:18]:
```suggestion
      GitHub API documentation: https://docs.github.com/en/rest
```

File: .github/ISSUE_TEMPLATE/api-request.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    options:
+    - "Yes"
+    - "No"
+- type: textarea
+  id: description
+  attributes:
+    label: api Feature Description
```
Comment (delvh) [inline .github/ISSUE_TEMPLATE/api-request.yml:25]:
```suggestion
    label: API Description
```

File: .github/ISSUE_TEMPLATE/api-request.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    placeholder: |
+      I think it would be great if Gitea had...
+  validations:
+    required: true
```
Comment (delvh) [inline .github/ISSUE_TEMPLATE/api-request.yml:29]: What about an extra section `Proposed routes` where you can add a list of all routes to add?
For example:
```md
- GET /api/v1/{user}/{repo}/{action}
- POST /api/v1/{user}/{repo}/{action}
```

File: .github/ISSUE_TEMPLATE/api-request.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+name: api feature Request
```
Comment (delvh) [inline .github/ISSUE_TEMPLATE/api-request.yml:1]:
```suggestion
name: API endpoint Request
```

**Comment by delvh** [comment]
Why did you close this PR?

**Pull request closed without merging**